src/main/java/de/hpi/msd/salsa/EdgeToAdjacencyApp.java

```diff
@@ -51,15 +62,46 @@
     @CommandLine.Option(names = "--buffer", defaultValue = "5000", description = "Buffer for reservoir sampling")
     private int bufferSize = 5000;
 
+    @CommandLine.Option(names = "--segments", defaultValue = "10", description = "Segments inside graphjet index")
+    private int segments = 10;
 
-    public final static String LEFT_INDEX_NAME = "leftIndex";
-    public final static String RIGHT_INDEX_NAME = "rightIndex";
-    private final Logger log = LoggerFactory.getLogger(EdgeToAdjacencyApp.class);
+    @CommandLine.Option(names = "--pools", defaultValue = "16", description = "Pools inside graphjet segment")
+    private int poolsPerSegment = 16;
+
+    @CommandLine.Option(names = "--nodesPerPool", defaultValue = "131072", description = "Nodes per graphjet pool")
+    private int nodesPerPool = 131072;
+
+    @Override
+    public Void call() throws Exception {
+        final Properties properties = this.getProperties();
+        final Topology topology = getTopology(properties, edgeProcessorType);
+        final KafkaStreams streams = new KafkaStreams(topology, properties);
 
+        streams.cleanUp();
+        streams.start();
+        waitForKafkaStreams(streams);
+
+        final BipartiteGraph graph = getGraph(edgeProcessorType, streams);
+        final RecommendationRestService recommendationRestService = new RecommendationRestService(graph);
+        final AdjacencyStateRestService adjacencyStateRestService = new AdjacencyStateRestService(graph);
+        final StreamsRestService restService = new StreamsRestService(new HostInfo(host, port),
+                recommendationRestService, adjacencyStateRestService);
+        restService.start();
+
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                streams.close();
+                restService.stop();
+            } catch (Exception e) {
+                log.warn("Error in shutdown", e);
+            }
+        }));
+        return null;
+    }
 
     public Properties getProperties() {
         Properties props = new Properties();
-        props.put(StreamsConfig.APPLICATION_ID_CONFIG, this.getClass().getSimpleName());
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, applicationId);
```

Comment:
Just something to keep in mind: We probably should still add a random string to the id to avoid Kafka Streams' auto-scaling